tests/flytekit/unit/core/test_workflows.py

```diff
@@ -16,7 +14,14 @@
 from flytekit.core.workflow import WorkflowFailurePolicy, WorkflowMetadata, WorkflowMetadataDefaults, workflow
 from flytekit.exceptions.user import FlyteValidationException, FlyteValueException
 from flytekit.tools.translator import get_serializable
-from flytekit.types.schema import FlyteSchema
+
+if typing.TYPE_CHECKING:
+    import pandas as pd
+else:
+    pd = pytest.importorskip("pandas")
+from flytekit.types.schema import FlyteSchema  # noqa: E402
```

Comment:
I think skipping this whole file when pandas is not installed is a bit too much. Many of the tests in this file does not look pandas specific.

This comment applies to some of the other tests that has `pd = pytest.importorskip("pandas")` at the top.

Reply:
Changed to `pytest.mark.skipif`. Thanks for the suggestion.

Reply:
except these files since most of tests use pandas in those files
<img width="861" alt="image" src="https://github.com/flyteorg/flytekit/assets/37936015/5b0bf9de-3e3e-4e15-8dd4-bd0f37ec01ec">